### PR TITLE
a bunch of fixes relating to shipping and brb login issue

### PR DESCRIPTION
**Pull request by AAAstorga**
- fixed brb login issues (for once and for all)
- added better error handling
- fixed issue where loading indicator would disappear too quickly

Once this is merged I will deploy new testflight 😄

### CODE REVIEW

File: Eatery Watch App Extension/TimeFactory.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import Foundation
+import DiningStack
+
+extension TimeInterval {
+    static func intervalWithHoursAndMinutesFromNow(hours: Double, minutes: Double) -> TimeInterval {
```
Comment (whoiswillma) [inline Eatery Watch App Extension/TimeFactory.swift:5]: This should be renamed interval(withHours hours: TimeInterval, minutes: TimeInterval) -> TimeInterval

File: Eatery Watch App Extension/TimeFactory.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+private func amOrPm(_ hour: Int) -> String {
+    if hour >= 12 {
+        return "pm"
+    } else {
+        return "am"
+    }
+}
+
+extension String {
+    mutating func convertTimeIfNeeded() {
```
Comment (whoiswillma) [inline Eatery Watch App Extension/TimeFactory.swift:78]: This feels strange as an extension on String. If this extension is not used anywhere else just including this logic inside dateConverter(_: Date, _: Date) or making this function fileprivate would be better.

File: Eatery/Controllers/EateriesViewController.swift [hunk 3]
```diff
@@ -209,6 +210,7 @@ class EateriesViewController: UIViewController, MenuButtonsDelegate, CLLocationM
     }
     
     func loadData(force: Bool, completion:(() -> Void)?) {
+        if initialLoad && !force { return }
```
Comment (whoiswillma) [inline Eatery/Controllers/EateriesViewController.swift:213]: Guard statement: 
`guard force || !initialLoad`

File: Eatery/Extensions/Eatery+Extension.swift [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                }
+            }
+        } else {
+            return .closed("Closed", "today")
+        }
+    }
+
+    // Retrieves a string list of the hours of operation for a day/time
+    func activeEventsForDate(date: Date) -> String {
+        var resultString = "Closed"
```
Comment (whoiswillma) [inline Eatery/Extensions/Eatery+Extension.swift:116]: Be more explicit about when this is "Closed". `if events.isEmpty`, else as such.

File: Eatery/Network/NetworkManager.swift [hunk 5]
```diff
@@ -94,14 +103,27 @@ struct NetworkManager {
...
-            guard let result = result, let data = result.data, let accountInfo = data.accountInfo else { return }
+            guard error == nil else {
+                completion(nil, NetworkError(message: error?.localizedDescription ?? ""))
+                return
+            }
+
+            guard let result = result,
+                let data = result.data,
+                let accountInfo = data.accountInfo else {
+                    completion(nil, NetworkError(message: "could not safely unwrap response"))
```
Comment (whoiswillma) [inline Eatery/Network/NetworkManager.swift:116]: Is NetworkError something that is presented to the user?

File: Eatery/Extensions/Eatery+Extension.swift [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    case closing(String, String)
+    case closed(String, String)
+}
+
+private let ShortDateFormatter: DateFormatter = {
+    let formatter = DateFormatter()
+    formatter.dateFormat = "h:mma"
+    return formatter
+}()
+private let kEateryAppendix = JSON(try! Data(contentsOf: Bundle.main.url(forResource: "appendix", withExtension: "json")!)).dictionaryValue
```
Comment (whoiswillma) [inline Eatery/Extensions/Eatery+Extension.swift:16]: Rename to `eateryAppendix`

**Review by whoiswillma — APPROVED** [review]

File: Eatery/Controllers/EateriesViewController.swift (hunk 3 above)
Comment (whoiswillma) [reply Eatery/Controllers/EateriesViewController.swift:213]: This whole method is kind of strange actually. Having an initial load method would be more clear.

File: Eatery/Extensions/Eatery+Extension.swift (hunk 6 above)
Comment (AAAstorga) [reply Eatery/Extensions/Eatery+Extension.swift:16]: I am going to leave this for now. This was in the previous versions, but I had to add it back for the watch app. Will eventually be removed

File: Eatery Watch App Extension/TimeFactory.swift (hunk 1 above)
Comment (whoiswillma) [reply Eatery Watch App Extension/TimeFactory.swift]: Renamed to `interval(withHours hours: TimeInterval, minutes: TimeInterval) -> TimeInterval`

File: Eatery/Network/NetworkManager.swift (hunk 5 above)
Comment (AAAstorga) [reply Eatery/Network/NetworkManager.swift:116]: No

**Pull request merged**